app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java

```diff
@@ -319,24 +319,39 @@ private LayoutParams getPluginsContainerLayoutParams() {
 
     private void setCompass() {
         mCompass = new Compass(getContext(), null, this);
+        mCompass.setOnClickListener(v -> rotateToNorth());
+
         mPluginsContainerLayout.addView(
-                mCompass, getRelativeLayoutParams(RelativeLayout.ALIGN_PARENT_TOP));
+                mCompass, getCompassLayoutParams(RelativeLayout.ALIGN_PARENT_TOP));
+
         mCompass.start();
     }
 
+    private void rotateToNorth() {
+        mLocationDisplayer.displaySelfLocation();
+        setRotationAngle(0);
+    }
+
+    private RelativeLayout.LayoutParams getCompassLayoutParams(int align) {
```

Comment:
Why "align" param is part of this method?
Extract the logic from here, this method should return the compass layout without doing anything else.